ejb3/src/main/java/org/jboss/as/ejb3/subsystem/ApplicationSecurityDomainService.java

```diff
@@ -120,10 +122,12 @@ public void cancel() {
 
         private final SecurityDomain securityDomain;
         private final boolean enableJacc;
+        private final boolean legacyCompliantPrincipalPropagation;
 
-        public ApplicationSecurityDomain(final SecurityDomain securityDomain, boolean enableJacc) {
+        public ApplicationSecurityDomain(final SecurityDomain securityDomain, boolean enableJacc, boolean legacyCompliantPrincipalPropagation) {
```

Comment:
this flag `legacyCompliantPrincipalPropagation` is passed from the enclosing class. Maybe we can just omit it and access the flag in the outer class directly.

Reply:
Fixed